internal/backend/db/dbgorm/values.go

```diff
@@ -0,0 +1,59 @@
+package dbgorm
+
+import (
+	"context"
+
+	"go.autokitteh.dev/autokitteh/internal/backend/db/dbgorm/scheme"
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+func (db *gormdb) SetValue(ctx context.Context, envID sdktypes.EnvID, key string, v sdktypes.Value) error {
+	if !envID.IsValid() {
+		return sdkerrors.NewInvalidArgumentError("invalid env_id")
+	}
+
+	j, err := v.MarshalJSON()
+	if err != nil {
+		return err
+	}
+
+	return translateError(db.db.Save(&scheme.Value{EnvID: envID.UUIDValue(), Key: key, Value: j}).Error)
+}
+
+func getValue(ctx context.Context, db *gormdb, envID sdktypes.EnvID, key string) (sdktypes.Value, error) {
+	r, err := getOne[scheme.Value](db.withUserEnvs(ctx), "env_id = ? AND name = ?", envID.String(), key)
+	if err != nil {
+		return sdktypes.InvalidValue, err
+	}
+
+	var v sdktypes.Value
+	if err := v.UnmarshalJSON(r.Value); err != nil {
+		return sdktypes.InvalidValue, err
+	}
+
+	return v, nil
+}
+
+func (db *gormdb) GetValue(ctx context.Context, envID sdktypes.EnvID, key string) (sdktypes.Value, error) {
+	return getValue(ctx, db, envID, key)
+}
+
+func (db *gormdb) ListValues(ctx context.Context, envID sdktypes.EnvID) (map[string]sdktypes.Value, error) {
+	// TODO: authz.
+
+	var rs []*scheme.Value
+	if err := db.db.WithContext(ctx).Where("env_id = ?", envID.UUIDValue()).Find(&rs).Error; err != nil {
```

Comment:
security issue

you don't check the user is allowed to access this field

Reply:
fixed